modules/stochastic_tools/include/utils/Calculators.h

```diff
@@ -43,80 +49,156 @@ class Calculator : public libMesh::ParallelObject
 {
 public:
   Calculator(const libMesh::ParallelObject & other, const std::string & name)
-    : libMesh::ParallelObject(other), _name(name)
+    : libMesh::ParallelObject(other), _name(name), _state(CalculatorState::NONE)
   {
   }
 
   virtual ~Calculator() = default;
-  virtual OutType compute(const InType &, bool) const = 0;
+  OutType compute(const InType &, bool);
+
+  void initializeCalculator();
+  void updateCalculator(const typename InType::value_type &);
+  void finalizeCalculator(bool);
```

Comment:
Can you add some documentation on these methods as well as on the pure virtual methods as well.

Reply:
Oops, yeah, definitely need to do that.